pkg/client/core/store/filesystem/application_filesystem.go

```diff
@@ -14,14 +16,47 @@ type applicationStore struct {
 	fs    *afero.Afero
 }
 
+// Helper for optional resources
+func addOperationIfNotEmpty(operations store.Operations, root, filePath string, content []byte) {
```

Comment:
We could probably add this to `store` by using `OperationOption` or something similar, but might not be worth the effort? Something to consider, I kind of like this functionality.

Reply:
We researched this option, but it takes too much effort right now.